### PR TITLE
Default memory unit to MiB while editing VM memory on details card

### LINKED ISSUE #999: When editing the amount of memory for a VM, memory units are automatically changed from MiB to GiB
When editing the amount of memory for a VM, memory units are automatically changed from MiB to GiB, which may lead to a VM which is not able to be scheduled

Version-Release number of selected component (if applicable):
4.3

How reproducible:
100%

Steps to Reproduce:
1. Provision a VM and edit the memory amount
2. Change 384 MiB (for example) to 512 MiB

Actual results:
MiB is changed to GiB

Expected results:
Units stay the same unless it's >1000 and makes sense to scale
——
Found here:

https://github.com/oVirt/ovirt-web-ui/commit/7f39063f#diff-1ed030b4b0f12b219c73b9e938e03286R363

Did we document this anywhere/ Assuming it's GB for templates is probably ok, but I'm not seeing a tooltip or anything. It's probably better to just assume it's MiB for 'normal' VMs, or we could put a little more effort into it and change:

if (isNumber(value) && ...) {
...
}
// Addition
else if (isParsableByte(value)) {
...
}

function isParsableByte(input) {
    types = {
        "MiB":
          ["MiB", "MB"],
        "GiB":
          ["GiB", "GB"]
    }
    matched = false

    Object.keys(types).forEach(function (size) {
        types[size].some(function(x) {
          if (input.match(x)) {
              matched = size
              return true
          }   
        }) 
    }) 
    console.log(matched)
    return matched 
}

Or, y'know, just default to MiB ;) Or document it

### PR DESCRIPTION
**Pull request by thaorell**
Fixes #999

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: src/components/VmDetails/cards/DetailsCard/index.js [hunk 1]
```diff
@@ -691,7 +690,8 @@ class DetailsCard extends React.Component {
     const osType = vm.getIn(['os', 'type'])
     const osId = operatingSystems && operatingSystems.find(os => os.get('name') === osType).get('id')
 
-    const { unit: memoryUnit, value: memorySize } = convertValue('B', vm.getIn(['memory', 'total']))
```
Comment (bond95) [inline src/components/VmDetails/cards/DetailsCard/index.js]: I think it will be better leave this line, and change units to `MiB` only in edit mode.

Comment (thaorell) [reply src/components/VmDetails/cards/DetailsCard/index.js]: this line causes all memory to be rounded to GB during edit. for example: if only edit mode is set to MiB,  1023, 1024 and 1050 MB are all rounded to 1GB, which is inaccurate

Comment (bond95) [reply src/components/VmDetails/cards/DetailsCard/index.js]: Yeah, but if we want to be 100% accurate we should use bytes =) Actually we already using formatting with precision in utilization cards, so you'll get not `1 GiB` for `1023 MiB`, but `0.99 GiB`.
https://github.com/oVirt/ovirt-web-ui/blob/e83d2f9357104f988a42168d7cd6ed6e0115cdea/src/components/VmDetails/cards/UtilizationCard/MemoryCharts.js#L39-L41

Comment (thaorell) [reply src/components/VmDetails/cards/DetailsCard/index.js]: Noted

File: src/components/VmDetails/cards/DetailsCard/index.js [hunk 2]
```diff
@@ -840,17 +840,17 @@ class DetailsCard extends React.Component {
                       }
                     </FieldRow>
                     <FieldRow label={msg.memory()} id={`${idPrefix}-memory`}>
-                      { !isFullEdit && `${round(memorySize)} ${memoryUnit}` }
+                      { !isFullEdit && `${round(memorySize / (1024 ** 3), 2)} ${'GiB'}` }
```
Comment (bond95) [inline src/components/VmDetails/cards/DetailsCard/index.js]: I meant not to hard-code `GiB`, but use function `userFormatOfBytes` to calculate most preferable unit with precision.

File: src/components/VmDetails/cards/DetailsCard/index.js [hunk 3]
```diff
@@ -840,17 +840,17 @@ class DetailsCard extends React.Component {
...
                           <FormControl
                             id={`${idPrefix}-memory-edit`}
                             className={style['memory-input']}
                             type='number'
-                            value={round(memorySize)}
+                            value={(memorySize / (1024 ** 2))}
                             onChange={e => this.handleChange('memory', e.target.value)}
                           />
-                          {memoryUnit}
+                          {'MiB'}
```
Comment (bond95) [inline src/components/VmDetails/cards/DetailsCard/index.js]: It not needed using brackets, you can just write:
```JavaScript
/>
MiB
```

File: src/components/VmDetails/cards/DetailsCard/index.js (hunk 2 above)
Comment (thaorell) [reply src/components/VmDetails/cards/DetailsCard/index.js]: didn't know this existed!

**Comment by isaranova** [comment]
LGTM

**Pull request merged**